com.microsoft.mrtk.graphicstools.unity/Editor/Utilities/InspectorUtilities.cs

```diff
@@ -11,10 +14,102 @@ namespace Microsoft.MixedReality.GraphicsTools.Editor
+                    return gameObject;
+                }
+            }
+
+            PlaceUIElementRoot.Invoke(null, new object[] { gameObject, menuCommand});
+
+            // The above call will create a new Canvas for us (if we don't have one),
+            // but it won't have optimal settings for MRTK UX. Let's fix that!
+            Canvas canvas = gameObject.GetComponentInParent<Canvas>();
+            RectTransform rt = canvas.GetComponent<RectTransform>();
+
+            // If the canvas's only child is us; let's make sure the Canvas has reasonable starting defaults.
+            // Otherwise, it was probably an existing canvas we were added to, so we shouldn't mess with it.
+            if (rt.childCount == 1 && rt.GetChild(0) == gameObject.transform)
+            {
+                SetReasonableCanvasDefaults(canvas);
+                
+                // Reset our own object to zero-position relative to the parent canvas.
+                gameObject.GetComponent<RectTransform>().anchoredPosition3D = Vector3.zero;
+            }
+
+            return gameObject;
+        }
+
+        /// <summary>
+        /// Configure some default properties for the canvas GameObject.
+        /// Size, position from camera
+        /// </summary>
+        private static void SetReasonableCanvasDefaults(Canvas canvas)
+        {
+            RectTransform rt = canvas.GetComponent<RectTransform>();
+
+            // 1mm : 1 unit measurement ratio.
+            if (rt.lossyScale != Vector3.one * 0.001f)
+            {
+                rt.localScale = Vector3.one * 0.001f;
+            }
+            // 150mm x 150mm.
+            rt.sizeDelta = Vector2.one * 150.0f;
+
+            // All our canvases will be worldspace (by default.)
+            canvas.renderMode = RenderMode.WorldSpace;
+            Undo.RecordObject(canvas, "Set Canvas RenderMode to WorldSpace");
+
+            // 30cm in front of the camera.
+            rt.position = Camera.main.transform.position + Camera.main.transform.forward * 0.3f;
+            Undo.RecordObject(rt, "Set Canvas Position");
+
+            // No GraphicRaycaster by default. Users can add one, if they like.
+            if (canvas.TryGetComponent(out GraphicRaycaster raycaster))
+            {
+                Undo.DestroyObjectImmediate(raycaster);
+            }
+
+            // CanvasScaler should be there by default.
+            if (!canvas.TryGetComponent(out CanvasScaler _))
+            {
+                Undo.AddComponent<CanvasScaler>(canvas.gameObject);
+            }
+        }
+
         /// <summary>
         /// Creates a new game object of type T as a child of the menu command.
         /// </summary>
-        public static GameObject CreateGameObjectFromMenu<T>(MenuCommand menuCommand) where T : MonoBehaviour
+        public static GameObject CreateGameObjectFromMenu<T>(MenuCommand menuCommand, Boolean hasCanvasParent = false) where T : MonoBehaviour
```

Comment:
Nit, there is nothing wrong with this, but most C# developers prefer the `bool` alias to `Boolean`
```suggestion
        public static GameObject CreateGameObjectFromMenu<T>(MenuCommand menuCommand, bool hasCanvasParent = false) where T : MonoBehaviour
```